plugins/modules/web_app_pool.ps1

```diff
@@ -0,0 +1,347 @@
+                            "Value: $new_value, Exception: $($_.Exception.Message)"
+                        )
+                        $module.FailJson($msg)
+                    }
+                }
+                $module.Result.changed = $true
+            }
+        }
+
+        # Set the state of the pool
+        if ($pool.State -eq "Stopped") {
+            if ($state -eq "started" -or $state -eq "restarted") {
+                if (-not $check_mode) {
+                    try {
+                        Start-WebAppPool -Name $name > $null
+                    }
+                    catch {
+                        $module.FailJson("Failed to start Web App Pool $($name): $($_.Exception.Message)")
+                    }
+                }
+                $module.Result.changed = $true
+            }
+        }
+        else {
+            if ($state -eq "stopped") {
+                if (-not $check_mode) {
+                    try {
+                        Stop-WebAppPool -Name $name > $null
+                    }
+                    catch {
+                        $module.FailJson("Failed to stop Web App Pool $($name): $($_.Exception.Message)")
+                    }
+                }
+                $module.Result.changed = $true
+            }
+            elseif ($state -eq "restarted") {
+                if (-not $check_mode) {
+                    try {
+                        Restart-WebAppPool -Name $name > $null
+                    }
+                    catch {
+                        $module.FailJson("Failed to restart Web App Pool $($name): $($_.Exception.Message)")
+                    }
+                }
+                $module.Result.changed = $true
+            }
+        }
+    }
+}
+
+# Get all the current attributes for the pool
+$pool = Get-Item -LiteralPath IIS:\AppPools\$name -ErrorAction SilentlyContinue
+$elements = @("attributes", "cpu", "failure", "processModel", "recycling")
+
+foreach ($element in $elements) {
+    if ($element -eq "attributes") {
+        $attribute_collection = $pool.Attributes
+        $attribute_parent = $pool
+    }
+    else {
+        $attribute_collection = $pool.$element.Attributes
+        $attribute_parent = $pool.$element
+    }
+
+    foreach ($attribute in $attribute_collection) {
+        $attribute_name = $attribute.Name
+        if ($attribute_name -notlike "*password*") {
+            $attribute_value = $attribute_parent.$attribute_name
+
+            $module.Result.info.$element.Add($attribute_name, $attribute_value)
+        }
+    }
+}
+
+# Manually get the periodicRestart attributes in recycling
+foreach ($attribute in $pool.recycling.periodicRestart.Attributes) {
+    $attribute_name = $attribute.Name
+    $attribute_value = $pool.recycling.periodicRestart.$attribute_name
+    $module.Result.info.recycling.periodicRestart.Add($attribute_name, $attribute_value)
+}
```

Comment:
If we decide to remove the return values in favour of a dedicated `_info` module then we can remove all this.